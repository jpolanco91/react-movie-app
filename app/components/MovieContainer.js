var React = require('react');
var MovieElement = require('./MovieElement');

class MovieContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      movies: []
    }
  }

  componentDidMount() {
    Axios.get('http://varnatrd.tech/api/series').then(({ data }) => {
      console.log(data)
      this.setState({ movies: data })
    })
  }

  render() {
    return (
      <div>
      {this.state.movies.map((movie) => {
        return (<MovieElement imageSource={movie.fullImage} title={movie.title} year={movie.year} />)
      })}
      </div>
    )
  }
}

module.exports =  MovieContainer;
