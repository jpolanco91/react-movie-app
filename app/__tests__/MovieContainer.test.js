import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MovieContainer from '../components/MovieContainer';

test('MovieContainer snapshot test', () => {
  const moviesMock = [
    {
      _id: '1',
      title: 'My title',
      year: 2008,
      genres: ['Crime', 'Thriller', 'Action'],
      covertImage: 'http://www.myimage.com/myImage.png',
      fullImage: 'http://www.myimage.com/myImage.png',
      trailer: 'http://www.mytrailer.com/trailer',
      rating: 9.5,
      categories: [],
      subTitle: 'Subtitle',
      isNew: false,
      isViewed: false,
      isFavorite: false,
      isRecent: false,
    },
  ];

  const c = create(
    <BrowserRouter>
      <MovieContainer movies={moviesMock} />
    </BrowserRouter>,
  );
  expect(c.toJSON()).toMatchSnapshot();
});
