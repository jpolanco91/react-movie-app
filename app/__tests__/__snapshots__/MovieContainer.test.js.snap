// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer snapshot test 1`] = `
<section
  className="section is-medium"
>
  <div
    className="container"
  >
    <div
      className="columns is-multiline"
    >
      <div
        className="column is-one-quarter"
      >
        <div
          className="card"
        >
          <div
            className="card-image"
          >
            <figure
              className="image is-4by3"
            >
              <img
                alt=""
                src="http://www.myimage.com/myImage.png"
              />
            </figure>
          </div>
          <div
            className="card-content"
          >
            <div
              className="media-content"
            >
              <a
                href="/movie_trailer/1"
                onClick={[Function]}
              >
                <p
                  className="title is-4"
                >
                  My title
                </p>
              </a>
              <p
                className="subtitle is-6"
              >
                2008
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
